drv/gimlet-hf-server/src/main.rs

```diff
@@ -147,11 +151,96 @@ fn main() -> ! {
         dev_select_pin: cfg.flash_dev_select,
     };
 
+    server.ensure_persistent_data_is_redundant().unwrap(); // TODO: log this?
```

Comment:
There's also #994; maybe not as part of this PR, but should we take Cliff's suggestion of "server starts in a degraded state" for both this failure and the capacity failure above?

Reply:
Separate thought - since we're scanning the flashes on startup anyway, should we remember the current data in `server`? I think that would allow `get_persistent_data()` to be idempotent (just return the current data), and `write_persistent_data()` could no-op out if there were no changes. The latter seems like it might be useful to protect against spurious flash writes if we had an automated process that was (e.g.) setting startup options on every boot (but changing them infrequently).

Reply:
I considered storing persistent data in the server, but decided it was extra complexity versus always doing the scans (which is Fast Enough).

However, "don't write if the persistent data matches" seems like a good plan; done in d6c8d269b9b45ce0c5c157e56d17a22e0056642b

Reply:
My "don't write if the data matches" suggest was intended for the non-persistent writing function (i.e., the idol call). I'm not sure who the client is going to be - probably control-plane-agent? - but it seems well within reason we could have something setting startup options to the same thing they're already set to, without realizing that could cause flash wear.

Reply:
Now that we're not storing host flash options at all (per discussions in #oxide-host-software), I think this is good to go?

Reply:
The host has ownership of the flash some of the time which means that the host can write to it. If the host is compromised, what mischief is possible?

Reply:
If the host could modify the flash, then it can already modify the phase1 image, so there's limited _new_ mischief that it could do.  A malicious host could write a header with a very high monotonic counter, which would supersede all other headers when deciding which header to use.  We'd see that represented by a `MonotonicCounterOverflow` error and could respond accordingly (e.g. erasing both flashes to reset `monotonic_counter = 0`).